test/Microsoft.AspNet.Razor.Test/Generator/CSharpTagHelperRenderingTest.cs

```diff
@@ -523,28 +600,42 @@ public static TheoryData RuntimeTimeTagHelperTestData
             {
                 // Test resource name, expected TagHelperDescriptors
                 // Note: The baseline resource name is equivalent to the test resource name.
-                return new TheoryData<string, IEnumerable<TagHelperDescriptor>>
+                return new TheoryData<string, string, IEnumerable<TagHelperDescriptor>>
                 {
-                    { "SingleTagHelper", DefaultPAndInputTagHelperDescriptors },
-                    { "BasicTagHelpers", DefaultPAndInputTagHelperDescriptors },
-                    { "BasicTagHelpers.RemoveTagHelper", DefaultPAndInputTagHelperDescriptors },
-                    { "BasicTagHelpers.Prefixed", PrefixedPAndInputTagHelperDescriptors },
-                    { "ComplexTagHelpers", DefaultPAndInputTagHelperDescriptors },
-                    { "DuplicateTargetTagHelper", DuplicateTargetTagHelperDescriptors },
-                    { "EmptyAttributeTagHelpers", DefaultPAndInputTagHelperDescriptors },
-                    { "EscapedTagHelpers", DefaultPAndInputTagHelperDescriptors },
-                    { "AttributeTargetingTagHelpers", AttributeTargetingTagHelperDescriptors },
+                    { "SingleTagHelper", null, DefaultPAndInputTagHelperDescriptors },
+                    { "BasicTagHelpers", null, DefaultPAndInputTagHelperDescriptors },
+                    { "BasicTagHelpers.RemoveTagHelper", null, DefaultPAndInputTagHelperDescriptors },
+                    { "BasicTagHelpers.Prefixed", null, PrefixedPAndInputTagHelperDescriptors },
+                    { "ComplexTagHelpers", null, DefaultPAndInputTagHelperDescriptors },
+                    { "DuplicateTargetTagHelper", null, DuplicateTargetTagHelperDescriptors },
+                    { "EmptyAttributeTagHelpers", null, DefaultPAndInputTagHelperDescriptors },
+                    { "EscapedTagHelpers", null, DefaultPAndInputTagHelperDescriptors },
+                    { "AttributeTargetingTagHelpers", null, AttributeTargetingTagHelperDescriptors },
+                    { "PrefixedAttributeTagHelpers", null, PrefixedAttributeTagHelperDescriptors },
+                    {
+                        "PrefixedAttributeTagHelpers",
+                        "PrefixedAttributeTagHelpers.Reversed",
+                        PrefixedAttributeTagHelperDescriptors.Reverse()
+                    },
                 };
             }
         }
 
         [Theory]
         [MemberData(nameof(RuntimeTimeTagHelperTestData))]
-        public void TagHelpers_GenerateExpectedRuntimeOutput(string testName,
-                                                             IEnumerable<TagHelperDescriptor> tagHelperDescriptors)
+        public void TagHelpers_GenerateExpectedRuntimeOutput(
```

Comment:
Run it through DesignTime too to validate that the prefix doesn't explode line mappings.